### PR TITLE
Better support for monorepos

### PR DESCRIPTION
**Pull request by zickgraf**
Various changes which improve the support for monorepos like https://github.com/homalg-project/homalg_project. The existing behavior for normal repos should be essentially unchanged.

See commit messages for details. Regarding the forth commit: This situation happens for our monorepos because our `package.yml` files are actually symlinks which point outside of WEB_DIR: https://github.com/homalg-project/homalg_project/blob/gh-pages/4ti2Interface/_data/package.yml

### CODE REVIEW

File: release-gap-package
Comment (fingolfin) [inline release-gap-package:198]: Maybe this would be more helpful?
```suggestion
    error "could not find ${UPDATE_FILE} file"
```

**Comment by fingolfin** [comment]
Why exactly are the package.yml outside and only symlinks in the WEBDIR? Can't you reverse that? Maybe Jekyll doesn't like symlinks in _data?

File: release-gap-package
Comment (fingolfin) [inline release-gap-package:149]: I am not thrilled by the name --update-file, but I also can't think of  good alternative right now, so... *shrug*

**Review by fingolfin — APPROVED** [review]
Overall this looks good to me, thanks! Only one minor optional suggestion. I'll wait for your reply before merging.

File: release-gap-package
Comment (zickgraf) [reply release-gap-package]: Good idea, I have pushed a slightly different suggestion (a841bcfb444b7d8687ddf6793e9390ba65a53440) which is also helpful in case one passes the empty string to --update-file.

**Comment by zickgraf** [comment]
> Why exactly are the package.yml outside and only symlinks in the WEBDIR? Can't you reverse that? Maybe Jekyll doesn't like symlinks in _data?

I did not try it, but even if GitHub Pages support symlinks at the moment I would not like to rely on this for the future if not necessary (since to me supporting symlinks sounds like something that might change without notice).

File: release-gap-package
Comment (zickgraf) [reply release-gap-package:149]: I think this name already existed in the old homalg_project version of the release script and I simply used it without thinking about it. Maybe "update-script" or "update-website-data-script" would be better?

Comment (fingolfin) [reply release-gap-package:149]: I think `--update-script` is marginally better. But in the end, it's bikeshedding. If you don't feel changing to update-script is to onerous, do it, otherwise, I'll also merge as-is *shrug*.

File: release-gap-package
Comment (fingolfin) [inline release-gap-package:198]: Thanks, that looks good

File: release-gap-package
Comment (zickgraf) [reply release-gap-package]: Done, see the last commit :-)

**Pull request merged**
